builder/frameworks/component_manager.py

```
@@ -0,0 +1,230 @@
+        """Remove CPPPATH entries for removed components from pioarduino-build.py."""
+        build_py_path = join(self.arduino_libs_mcu, "pioarduino-build.py")
+        
+        if not os.path.exists(build_py_path):
+            return
+        
+        with open(build_py_path, 'r') as f:
+            content = f.read()
+        
+        # Remove CPPPATH entries for each removed component
+        for component in self.removed_components:
+            escaped_component = re.escape(component_name)
```

Ruff (0.11.9)

Undefined name `component_name`

(F821)